Build Error: ext_assimp patch failed due to missing 0003-fix-draco-cmake-intelllvm-cxx-flags.patch

### Checklist

- [x] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [x] For Python issues, I have tested with the [latest development wheel](https://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [x] I have checked the [release documentation](https://www.open3d.org/docs/release/) and the [latest documentation](https://www.open3d.org/docs/latest/) (for `main` branch).

### Steps to reproduce the issue

#### I first cloned Open3D by:
```
git clone https://github.com/isl-org/Open3D.git
cd Open3D
```

#### Then, I build Open3D (on Ubuntu 24.04, without cuda) with:
```
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/Project/depends -DBUILD_PYTHON_MODULE=OFF 
make -j$(nproc)
```
Build Error: ext_assimp patch failed due to missing 0003-fix-draco-cmake-intelllvm-cxx-flags.patch.
the build process fails during the ext_assimp compilation phase. CMake's ExternalProject_Add tries to apply a third-party patch named 0003-fix-draco-cmake-intelllvm-cxx-flags.patch, but the file does not exist in the 3rdparty/assimp/ directory.

Upon checking the local repository and the official upstream repository, only 0001-...patch and 0002-...patch are present under 3rdparty/assimp/. The missing 0003 patch causes the git apply command to fail with exit code 128, halting the entire build.

### Error message

```shell
[ 23%] Performing patch step for 'ext_assimp'
Initialized empty Git repository in /home/cas/Project/Open3D/build/assimp/src/ext_assimp/.git/
error: can't open patch '/home/cas/Project/Open3D/3rdparty/assimp/0003-fix-draco-cmake-intelllvm-cxx-flags.patch': No such file or directory
make[2]: *** [CMakeFiles/ext_assimp.dir/build.make:120: assimp/src/ext_assimp-stamp/ext_assimp-patch] Error 128
make[1]: *** [CMakeFiles/Makefile2:909: CMakeFiles/ext_assimp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Open3D, Python and System information

```markdown
- Operating system: Ubuntu 24.04 
- Python version: Python 3.12
```

### Additional information

_No response_

--- a/3rdparty/assimp/0003-fix-draco-cmake-intelllvm-cxx-flags.patch
+++ b/3rdparty/assimp/0003-fix-draco-cmake-intelllvm-cxx-flags.patch
@@ -1,0 +1,39 @@
+--- a/contrib/draco/cmake/draco_flags.cmake
++++ b/contrib/draco/cmake/draco_flags.cmake
+@@ -226,14 +226,14 @@ endmacro()
+ macro(draco_set_cxx_flags)
+   unset(cxx_flag_lists)
+ 
+-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
++  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU|IntelLLVM")
+     list(APPEND cxx_flag_lists draco_base_cxx_flags)
+   endif()
+ 
+   # Append clang flags after the base set to allow -Wno* overrides to take
+   # effect. Some of the base flags may enable a large set of warnings, e.g.,
+   # -Wall.
+-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
++  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|IntelLLVM")
+     list(APPEND cxx_flag_lists draco_clang_cxx_flags)
+   endif()
+ 
+--- a/contrib/draco/cmake/draco_build_definitions.cmake
++++ b/contrib/draco/cmake/draco_build_definitions.cmake
+@@ -96,7 +96,7 @@ macro(draco_set_build_definitions)
+     endif()
+ 
+     if(NOT DRACO_DEBUG_COMPILER_WARNINGS)
+-      if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
++      if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|IntelLLVM")
+         list(APPEND draco_clang_cxx_flags
+                     "-Wno-implicit-const-int-float-conversion")
+       else()
+@@ -113,7 +113,7 @@ macro(draco_set_build_definitions)
+ 
+   set_draco_target()
+ 
+-  if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
++  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|IntelLLVM")
+     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "6")
+       # Quiet warnings in copy-list-initialization where {} elision has always
+       # been allowed.

